### PR TITLE
Combined Three Parts of the Model Into Single Code

### PR DESCRIPTION
**Pull request by anug8628**
Combined all three parts of the code to one. Given an input image, a spotify playlist is created

### CODE REVIEW

File: run_playlist_generator.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    image_path = "adolphe-joseph-thomas-monticelli_rural-scene.jpg"
+    caption = get_caption(image_path)
+    print(caption)
+
+    # step 2: generate mood based on image + caption
+    class_id = determine_mood(image_path, caption)
+    emotion = idx_to_emotion[class_id]
+    print(emotion)
+
+    # step 3: generate spotify playlist based on moods list
```
Comment (allison-liu) [inline run_playlist_generator.py:107]: nit: maybe would be easier to make step 3 its own function.. i only say this because im planning on adding more stuff for creating playlist/descriptions, which could make this code slightly longer

**Review by allison-liu — COMMENTED** [review]
super easy to understand your code thank you! just added one nit

File: run_playlist_generator.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+    # get predicted mood
+    logits = outputs['logits']
+    predicted_class_id = logits.argmax(-1).item()
+
+    return predicted_class_id
+
+
+def main():
+    # step 1: generate caption based on images
```
Comment (ruthleee) [inline run_playlist_generator.py:97]: Could you make it so that it takes in like 4-9 images? That way we have more moods to work with for the playlist and does the actual mood board part?

File: run_playlist_generator.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+from transformers import VisionEncoderDecoderModel, GPT2TokenizerFast, ViTImageProcessor
+
+# set device to GPU if available
+device = "cuda" if torch.cuda.is_available() else "cpu"
+
+caption_model = VisionEncoderDecoderModel.from_pretrained("nlpconnect/vit-gpt2-image-captioning").to(device)
+caption_tokenizer = GPT2TokenizerFast.from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+caption_image_processor = ViTImageProcessor.from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+
+def get_caption(image_path):
```
Comment (ruthleee) [inline run_playlist_generator.py:24]: I would add a little comment just for documentation to say that we use a different pretrained model from the one we built for accuracy purposes in having the end to end framework?

**Review by ruthleee — COMMENTED** [review]
Overall great work, Anushka! It would be great to see like a final script that where when it's run, it asks the user for like 9 images and then begins to generate the playlist. Some sort of very simple user interactive script. Thanks!

File: run_playlist_generator.py (hunk 1 above)
Comment (anug8628) [reply run_playlist_generator.py:107]: ok makes sense, i will change it to its own function

File: run_playlist_generator.py (hunk 2 above)
Comment (anug8628) [reply run_playlist_generator.py:97]: Yup! Will add the test to take in multiple images, especially since we need it for the analysis of the models
